src/coreclr/jit/lower.cpp

```diff
@@ -140,32 +135,35 @@ bool Lowering::CheckImmedAndMakeContained(GenTree* parentNode, GenTree* childNod
 }
 
 //------------------------------------------------------------------------
-// IsSafeToContainMem: Checks for conflicts between childNode and parentNode,
-// and returns 'true' iff memory operand childNode can be contained in parentNode.
+// IsInvariantInRange: Check if a node is invariant in the specified range. In
+// other words, can 'node' be moved to right before 'endExclusive' without its
+// computation changing values?
 //
 // Arguments:
-//    parentNode - any non-leaf node
-//    childNode  - some node that is an input to `parentNode`
+//    node         -  The node.
+//    endExclusive -  The exclusive end of the range to check invariance for.
 //
-// Return value:
-//    true if it is safe to make childNode a contained memory operand.
+// Returns:
+//    True if 'node' can be evaluated at any point between its current
+//    location and 'parentNode' without giving a different result; otherwise
+//    false.
 //
-bool Lowering::IsSafeToContainMem(GenTree* parentNode, GenTree* childNode) const
+bool Lowering::IsInvariantInRange(GenTree* node, GenTree* endExclusive) const
```

Comment:
Just wanted to comment that I really like the name change and parameter reorder.

Not only is it clearer what the check is doing (particularly since this didn't really have to do with memory), but its now clearly "check from node to endExclusive" and so you can think of it simply as "start, end+1"

Reply:
That being said, given the reorder it might be nice to assert that `node` precedes `endExclusive` in the linear order.

Would help prevent any accidents or other problems in the future.

Reply:
Added some asserts to make sure things look ok